include/aikido/planner/vectorfield/VectorFieldUtil.hpp

```diff
@@ -0,0 +1,79 @@
+#ifndef AIKIDO_PLANNER_VECTORFIELD_UTIL_H_
+#define AIKIDO_PLANNER_VECTORFIELD_UTIL_H_
+
+#include <dart/dynamics/BodyNode.hpp>
+#include <dart/optimizer/Function.hpp>
+#include <dart/optimizer/Solver.hpp>
+#include <dart/optimizer/nlopt/NloptSolver.hpp>
+#include <aikido/common/Spline.hpp>
+#include <aikido/statespace/dart/MetaSkeletonStateSpace.hpp>
+#include <aikido/trajectory/Spline.hpp>
+
+namespace aikido {
+namespace planner {
+namespace vectorfield {
+
+struct Knot
+{
+  double t;
+  Eigen::Matrix<double, 2, Eigen::Dynamic> values;
+};
+
+/// convert a sequence of knots into a Spline trajectory.
+/// \param knots a sequence of knots
+/// \param cache_index total cache index number
+/// \param stateSpace MetaSkeleton state space
+/// \return A Spline trajectory
+std::unique_ptr<aikido::trajectory::Spline> convertToSpline(
+    const std::vector<Knot>& knots,
+    ptrdiff_t cache_index,
+    aikido::statespace::dart::MetaSkeletonStateSpacePtr stateSpace);
+
+/// A function class that defines an objective. The objective measures
+/// the difference between a desired twist and Jacobian * joint velocities.
+class DesiredTwistFunction : public dart::optimizer::Function
+{
+public:
+  using Twist = Eigen::Matrix<double, 6, 1>;
+  using Jacobian = dart::math::Jacobian;
+
+  /// Constructor.
+  /// \param _twist a desired twist
+  /// \param _jacobian system Jacobian
+  DesiredTwistFunction(const Twist& _twist, const Jacobian& _jacobian);
```

Comment:
Nit: I believe the convention is not to use `_` (underscore) prefix for parameter names.